### PR TITLE
README.md: Move all mentions of release tarballs to the installation guide

### LINKED ISSUE #30201: Developer's Guide: Update git installation instructions
<div id="comment:0"></div>

https://doc.sagemath.org/html/en/developer/git_setup.html#installing-git

On mac, it should refer to Xcode.
The linux install instructions should be autogenerated.

Depends on #29655

CC:  @jhpalmieri @orlitzky @fchapoton @dimpase @Etn40ff

Component: **documentation**

Keywords: **sd111**

_Issue created by migration from https://trac.sagemath.org/ticket/30201_

### PR DESCRIPTION
**Pull request by mkoeppe**
As proposed in https://groups.google.com/g/sage-devel/c/5kmxaw105lg/m/liCPKG4bAQAJ

The README.md now starts with instructions to install git and then use `git clone`.

- Also fixes https://github.com/sagemath/sage/issues/30201 

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/doc/en/installation/source.rst [hunk 1]
```diff
@@ -387,9 +387,49 @@ Installation steps
 #. Follow the procedure in the file `README.md <https://github.com/sagemath/sage/#readme>`_
    in ``SAGE_ROOT``.
 
+#. If you need wish to prepare for having to build Sage in an environment
```
Comment (jhpalmieri) [inline src/doc/en/installation/source.rst]: ``
```suggestion
#. If you wish to prepare for having to build Sage in an environment
```

Comment (jhpalmieri) [reply src/doc/en/installation/source.rst]: (Or delete "wish" and keep "need")

File: src/doc/en/installation/source.rst [hunk 2]
```diff
@@ -387,9 +387,49 @@ Installation steps
...
+     in the Release Assets, which contains a prepopulated subdirectory
+     ``upstream``.
+
+     After downloading the source tarball ```sage-x.y.tar.gz``` into
+     a directory ```~/sage/```::
+
+       $ cd ~/sage/
+       $ tar xf sage-x.y.tar.gz  # adapt x.y; takes a while
+
+     This creates the subdirectory `````sage-x.y```. Now change into it::
```
Comment (jhpalmieri) [inline src/doc/en/installation/source.rst]:
```suggestion
     This creates the subdirectory ``sage-x.y``. Now change into it::
```

File: src/doc/en/installation/source.rst [hunk 3]
```diff
@@ -387,9 +387,49 @@ Installation steps
...
+
+   - Alternatively, instead of cloning the git repository, you
+     can download a self-contained release tarball for any
+     stable release from the Sage project's
+     `GitHub Releases <https://github.com/sagemath/sage/releases>`_.
+     Use the file named ``sage-x.y.tar.gz`` (1.25 GB as of Sage 10.2)
+     in the Release Assets, which contains a prepopulated subdirectory
+     ``upstream``.
+
+     After downloading the source tarball ```sage-x.y.tar.gz``` into
```
Comment (jhpalmieri) [inline src/doc/en/installation/source.rst]:
```suggestion
     After downloading the source tarball ``sage-x.y.tar.gz`` into
```

Comment (jhpalmieri) [reply src/doc/en/installation/source.rst]: I don't think we need visible backticks in the html documentation, which is what triple backticks produce.

File: src/doc/en/installation/source.rst [hunk 4]
```diff
@@ -387,9 +387,49 @@ Installation steps
...
+     a directory ```~/sage/```::
```
Comment (jhpalmieri) [inline src/doc/en/installation/source.rst]:
```suggestion
     a directory ``~/sage/``::
```

**Review by jhpalmieri — COMMENTED** [review]
Looks good to me, just a few minor comments. I'm happy to approve, although I'm not sure if this should be merged yet, since the "discussion" on sage-devel is ongoing.

File: src/doc/en/installation/source.rst (hunk 3 above)
Comment (mkoeppe) [reply src/doc/en/installation/source.rst]: Thanks. I think I may need to clean my keyboard if these typos become more frequent

**Comment by mkoeppe** [comment]
> I'm not sure if this should be merged yet, since the "discussion" on sage-devel is ongoing.

I'd say, let's just go ahead. Haven't seen any reaction in the thread (other than yours)

**Comment by saraedum** [comment]
Cloning the repo uses 1GB of disk (without any upstream packages downloaded.) Using the tarball seems more economical to me in this case. Downloading a tarball of the GitHub repo (which is not advertised anywhere) is just 28MB for me.

Anyway, I don't really mind much either way. If you guys think this is a good idea, go ahead.

**Comment by dimpase** [comment]
> Cloning the repo uses 1GB of disk (without any upstream packages downloaded.) 

no, it's "only" 615 Mb, and over 60% of it is `.git/`
```
$ git clone -c core.symlinks=true --origin upstream --branch develop --tags https://github.com/sagemath/sage.git
...
$ du -h sage/
...
615M	sage/
$ du -h sage/.git
388M	sage/.git/objects/pack
...
390M	sage/.git
```

Adding `--filter=blob:none`, as recommended in https://github.blog/2020-12-21-get-up-to-speed-with-partial-clone-and-shallow-clone/ - results in only 141Mb in  `sage/.git`, and 366M for the whole `sage/`.

Perhaps this is, disk space and bandwidth-wise, a better advice than the one in the PR, to use the latter.


> Using the tarball seems more economical to me in this case. Downloading a tarball of the GitHub repo (which is not advertised anywhere) is just 28MB for me.
> 
> Anyway, I don't really mind much either way. If you guys think this is a good idea, go ahead.

**Review by dimpase — CHANGES_REQUESTED** [review]
consider adding `--filter=blob:none` to git clone options

**Comment by mkoeppe** [comment]
Done in 04145de

**Comment by mkoeppe** [comment]
Also need to link to git install instructions (similar to #30201)

**Comment by mkoeppe** [comment]
Thanks for the review.

**Pull request merged**